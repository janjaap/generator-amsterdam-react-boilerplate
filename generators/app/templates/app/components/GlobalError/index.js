import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import CloseIcon from '@datapunt/asc-assets/lib/Icons/Close.svg';
import AlertIcon from '@datapunt/asc-assets/lib/Icons/Alert.svg';

const ErrorWrapper = styled.div`
  background-color: #ec0000;
  color: white;
`;

const ErrorContainer = styled.div`
  max-width: 1024px;
  padding: 11px 16px;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
`;

const P = styled.p`
  margin: 0;
  max-width: calc(100% - 60px);
  flex: 1;
`;

const Button = styled.button`
  height: 25px;
  margin-left: 10px;
  width: 25px;
  -webkit-appearance: none;
  appearance: none;
  border: 0;
  float: right;
  background: transparent;
  cursor: pointer;
`;

const Label = styled.span`
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
`;

const GlobalError = ({ onClose, errorMessageLabel, closeLabel, feedbackLabel, showReportDialog }) => (
  <ErrorWrapper className="no-print cf">
    <ErrorContainer>
      <P>{errorMessageLabel}</P>
      {showReportDialog && (
        <Button type="button" onClick={showReportDialog} data-testid="globalerror-showReportDialog">
          <AlertIcon focusable="false" width={20} fill="#fff" />
          <Label>{feedbackLabel}</Label>
        </Button>
      )}
      <Button type="button" onClick={onClose}>
        <CloseIcon focusable="false" width={20} fill="#fff" />
        <Label>{closeLabel}</Label>
      </Button>
    </ErrorContainer>
  </ErrorWrapper>
);

GlobalError.defaultProps = {
  onClose: null,
  showReportDialog: null,
};

GlobalError.propTypes = {
  errorMessageLabel: PropTypes.string.isRequired,
  closeLabel: PropTypes.string.isRequired,
  feedbackLabel: PropTypes.string.isRequired,
  onClose: PropTypes.func,
  showReportDialog: PropTypes.func,
};

export default GlobalError;
