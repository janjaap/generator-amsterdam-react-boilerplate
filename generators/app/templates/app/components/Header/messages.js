import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
    defaultMessage: 'Logged in as',
  },
  log_in: {
    id: `${scope}.log_in`,
    defaultMessage: 'Log in',
  },
  log_out: {
    id: `${scope}.log_out`,
    defaultMessage: 'Log out',
  },
});
