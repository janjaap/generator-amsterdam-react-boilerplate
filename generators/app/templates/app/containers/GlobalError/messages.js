import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
  unknown_error: {
    id: `${scope}.unknown_error`,
  },
  server_error: {
    id: `${scope}.server_error`,
  },
  service_unavailable: {
    id: `${scope}.service_unavailable`,
  },
  resource_not_found: {
    id: `${scope}.resource_not_found`,
  },
  no_vbo_data_available: {
    id: `${scope}.no_vbo_data_available`,
  },
  login_failed: {
    id: `${scope}.login_failed`,
  },
  logout_failed: {
    id: `${scope}.logout_failed`,
  },
  authorize_failed: {
    id: `${scope}.authorize_failed`,
  },
  error_reporting_title: {
    id: `${scope}.error_reporting_title`,
  },
  error_reporting_subtitle: {
    id: `${scope}.error_reporting_subtitle`,
  },
  error_reporting_label_name: {
    id: `${scope}.error_reporting_label_name`,
  },
  error_reporting_label_email: {
    id: `${scope}.error_reporting_label_email`,
  },
  error_reporting_label_comments: {
    id: `${scope}.error_reporting_label_comments`,
  },
  error_reporting_label_close: {
    id: `${scope}.error_reporting_label_close`,
  },
  error_reporting_label_submit: {
    id: `${scope}.error_reporting_label_submit`,
  },
  error_reporting_success_message: {
    id: `${scope}.error_reporting_success_message`,
  },
  error_reporting_error_generic: {
    id: `${scope}.error_reporting_error_generic`,
  },
  error_reporting_error_form: {
    id: `${scope}.error_reporting_error_form`,
  },
});
