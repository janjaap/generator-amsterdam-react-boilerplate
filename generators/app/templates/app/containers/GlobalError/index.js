import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose, bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { injectIntl, intlShape } from 'react-intl';
import * as Sentry from '@sentry/browser';

import { makeSelectError, makeSelectErrorMessage, makeSelectErrorEventId } from 'containers/App/selectors';
import appMessages from 'containers/App/messages';
import { resetGlobalError } from 'containers/App/actions';

import GlobalError from 'components/GlobalError';

import errorMessages from './messages';

export const showReportDialog = (eventId, intl) => () => {
  Sentry.showReportDialog({
    eventId,
    title: intl.formatMessage(errorMessages.error_reporting_title),
    subtitle: intl.formatMessage(errorMessages.error_reporting_subtitle),
    subtitle2: '',
    labelName: intl.formatMessage(errorMessages.error_reporting_label_name),
    labelEmail: intl.formatMessage(errorMessages.error_reporting_label_email),
    labelComments: intl.formatMessage(errorMessages.error_reporting_label_comments),
    labelClose: intl.formatMessage(errorMessages.error_reporting_label_close),
    labelSubmit: intl.formatMessage(errorMessages.error_reporting_label_submit),
    errorGeneric: intl.formatMessage(errorMessages.error_reporting_error_generic),
    errorFormEntry: intl.formatMessage(errorMessages.error_reporting_error_form),
    successMessage: intl.formatMessage(errorMessages.error_reporting_success_message),
  });
};

export const GlobalErrorContainer = ({ error, errorMessage, errorEventId, intl, onClose }) => {
  if (!error) {
    return null;
  }

  const props = {
    errorMessageLabel: intl.formatMessage(errorMessages[errorMessage]),
    closeLabel: intl.formatMessage(appMessages.close),
    feedbackLabel: intl.formatMessage(appMessages.report_feedback),
    onClose,
    showReportDialog: errorEventId ? showReportDialog(errorEventId, intl) : null,
  };

  return <GlobalError {...props} />;
};

GlobalErrorContainer.defaultProps = {
  error: false,
  errorEventId: undefined,
  errorMessage: '',
  onClose: null,
};

GlobalErrorContainer.propTypes = {
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  intl: intlShape.isRequired,
  onClose: PropTypes.func,
  errorEventId: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  error: makeSelectError,
  errorMessage: makeSelectErrorMessage,
  errorEventId: makeSelectErrorEventId,
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      onClose: resetGlobalError,
    },
    dispatch,
  );

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const composed = compose(
  injectIntl,
  withConnect,
)(GlobalErrorContainer);

export { composed as default, GlobalError as GlobalErrorComponent };
