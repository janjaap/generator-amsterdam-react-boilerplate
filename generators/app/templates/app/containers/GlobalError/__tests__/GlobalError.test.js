import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import history from 'utils/history';
import { Provider } from 'react-redux';

import { intl } from '../../../../internals/testing/test-utils';
import GlobalError, { GlobalErrorContainer } from '..';
import messages from '../messages';
import configureStore from '../../../configureStore';

const store = configureStore({}, history);

describe('containers/GlobalError', () => {
  afterEach(cleanup);

  const intlObj = intl({ messages });

  it('should render showing no error by default', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <GlobalError />
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render showing an error when defined', () => {
    const props = {
      error: true,
      errorMessage: 'MOCK_ERROR',
      intl: intlObj,
    };
    const { container } = render(<GlobalErrorContainer {...props} errorMessage="unauthorized" />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.classList.contains('no-print')).toEqual(true);
  });

  it('should capture click on close button', () => {
    const onClose = jest.fn();
    const props = {
      error: true,
      errorMessage: 'MOCK_ERROR',
      onClose,
      intl: intlObj,
    };

    render(<GlobalErrorContainer {...props} errorMessage="unauthorized" />);

    const button = document.getElementsByTagName('button')[0];
    fireEvent.click(button);

    expect(onClose).toHaveBeenCalled();
  });
});
