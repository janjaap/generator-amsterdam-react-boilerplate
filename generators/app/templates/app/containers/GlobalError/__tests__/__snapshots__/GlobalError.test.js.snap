// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/GlobalError should render showing an error when defined 1`] = `
<div
  class="no-print cf GlobalError__ErrorWrapper-igrx8k-0 kuoiLJ"
>
  <div
    class="GlobalError__ErrorContainer-igrx8k-1 dDcAQy"
  >
    <p
      class="GlobalError__P-igrx8k-2 fGvvtl"
    >
      registraties.unauthorized
    </p>
    <button
      class="GlobalError__Button-igrx8k-3 dCWsfn"
      type="button"
    >
      <img
        fill="#fff"
        focusable="false"
        width="20"
      />
      <span
        class="GlobalError__Label-igrx8k-4 fSyAoo"
      >
        registraties.close
      </span>
    </button>
  </div>
</div>
`;

exports[`containers/GlobalError should render showing no error by default 1`] = `<div />`;
